lewis/devices/julabo/interfaces/julabo_stream_interface_2.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# *********************************************************************
+# plankton - a library for creating hardware device simulators
+# Copyright (C) 2016 European Spallation Source ERIC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# *********************************************************************
+
+from lewis.adapters.stream import StreamAdapter, Cmd
+
+
+class JulaboStreamInterfaceV2(StreamAdapter):
+    """Julabos can have different commands sets depending on the version number of the hardware.
+
+    This protocol matches that for: FP50-HE (unconfirmed).
+    """
+
+    protocol = "julabo-version-2"
+
+    commands = {
+        Cmd('get_bath_temperature', '^IN_PV_00$'),
+        Cmd('get_external_temperature', '^IN_PV_01$'),
+        Cmd('get_power', '^IN_PV_02$'),
+        Cmd('get_set_point', '^IN_SP_00$'),
+        Cmd('get_high_limit', '^IN_SP_03$'),    # Different from version 1
```

Comment:
Var is really cool - it has reduced the amount of code significantly.

As a suggestion it would be really nice is if you could pass Var limits to check then I could get rid of the setters that do nothing other than check the limits, like:

```
@check_limits(0, 999)
    def set_external_d(self, param):
        """
        Sets the external derivative.
        Tv in Julabo speak.

        :param param: The value to set, must be an integer between 0 and 999
        :return: Empty string.
        """
        self.external_d = param
        return ""
```

It might not be much use generally, so feel free to disregard.